lib/src/block_syntaxes/html_block_syntax.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../ast.dart';
+import '../block_parser.dart';
+import '../patterns.dart';
+import 'block_syntax.dart';
+
+/// Parse HTML blocks.
+class HtmlBlockSyntax extends BlockSyntax {
+  @override
+  RegExp get pattern => htmlBlockPattern;
+
+  // All types of HTML blocks except type 7 may interrupt a paragraph.
```

Comment:
This comment needs some context too. The "type 7" part. Maybe expand here, or link to commonmark, or add context to the comment above on line 10. Just because this is the first mention of "type 7" when reading the file from top to bottom.

Reply:
Done!